### PR TITLE
AA-279 M-04 Inconsistent Price Precision

### PR DESCRIPTION
**Pull request by shahafn**
Fixing priceUpdateThreshold

### CODE REVIEW

File: contracts/samples/utils/OracleHelper.sol [hunk 1]
```diff
@@ -39,7 +39,7 @@ abstract contract OracleHelper {
         /// @notice 'false' if price is dollars-per-ether, 'true' if price is ether-per-dollar
         bool nativeOracleReverse;
 
-        /// @notice The price update threshold percentage that triggers a price update (1e6 = 100%)
+        /// @notice The price update threshold percentage from PRICE_DENOMINATOR that triggers a price update (1e3 = 100%)
```
Comment (forshtat) [inline contracts/samples/utils/OracleHelper.sol]: I think `1e3` is better written as `1000`

File: contracts/samples/utils/OracleHelper.sol [hunk 2]
```diff
@@ -100,11 +100,10 @@ abstract contract OracleHelper {
             oracleHelperConfig.tokenOracleReverse,
             oracleHelperConfig.nativeOracleReverse
         );
-        uint256 priceNewByOld = price * PRICE_DENOMINATOR / _cachedPrice;
-
+        uint256 priceRatioNewByOld = price * PRICE_DENOMINATOR / _cachedPrice;
         bool updateRequired = force ||
-            priceNewByOld > PRICE_DENOMINATOR + priceUpdateThreshold ||
-            priceNewByOld < PRICE_DENOMINATOR - priceUpdateThreshold;
+            priceRatioNewByOld > PRICE_DENOMINATOR * (1 + priceUpdateThreshold / 1e3) ||
```
Comment (forshtat) [inline contracts/samples/utils/OracleHelper.sol]: I think you will just get `priceUpdateThreshold / 1e3 == 0` for any `priceUpdateThreshold < 1e3`, am I missing something?

**Review by forshtat — CHANGES_REQUESTED** [review]

File: contracts/samples/utils/OracleHelper.sol (hunk 1 above)
Comment (shahafn) [reply contracts/samples/utils/OracleHelper.sol]: Fixed

File: contracts/samples/utils/OracleHelper.sol (hunk 2 above)
Comment (shahafn) [reply contracts/samples/utils/OracleHelper.sol]: Indeed. There was an issue before as well. I think I fixed it.

File: contracts/samples/TokenPaymaster.sol [hunk 3]
```diff
@@ -26,7 +26,7 @@ import "./utils/OracleHelper.sol";
 contract TokenPaymaster is BasePaymaster, UniswapHelper, OracleHelper {
 
     struct TokenPaymasterConfig {
-        /// @notice The price markup percentage applied to the token price (1e6 = 100%)
+        /// @notice The price markup percentage applied to the token price (1e26 = 100%). Ranges from 1e26 to 2e26
```
Comment (drortirosh) [inline contracts/samples/TokenPaymaster.sol:29]: if 1e26 is 100%, then the range is 100% .. 200% ?

Comment (forshtat) [reply contracts/samples/TokenPaymaster.sol:29]: I think yes. This is the markup, so using 100% of the price is 0% markup and using 200% of the price means 100% markup, does that make sense?

**Review by forshtat — DISMISSED** [review]

**Review by drortirosh — APPROVED** [review]

**Pull request merged**